armi/interfaces.py

```diff
@@ -77,6 +80,117 @@ class STACK_ORDER:  # pylint: disable=invalid-name, too-few-public-methods
     POSTPROCESSING = BOOKKEEPING + 1
 
 
+class TightCoupler:
+    """
+    Data structure that defines tight coupling attributes that are implemented
+    within an Interface and called upon when ``interactCoupled`` is called.
+
+    Parameters
+    ----------
+    param : str
+        The name of a parameter defined in the ARMI Reactor model.
+
+    tolerance : float
+        Defines the allowable error, epsilon, between the current previous
+        parameter value(s) to determine if the selected coupling parameter has
+        been converged.
+
+    maxIters : int
+        Maximum number of iterations allowed before the ``isConverged`` method
+        will return True and the state of the object will be cleared.
+    """
+
+    _SUPPORTED_TYPES = [float, int, list, numpy.ndarray]
+
+    def __init__(self, param, tolerance, maxIters):
+        self.param = param
```

Comment:
Oops - This should be `self.parameter`. Still some cleanups to do.

Reply:
https://github.com/terrapower/armi/pull/1033/commits/914599da7c18d3a224147501b436c8fa5517cb46